code/android-signal-library/src/main/java/com/adobe/marketing/mobile/signal/EventExtensions.kt

```diff
@@ -0,0 +1,123 @@
+/*
+  Copyright 2022 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+package com.adobe.marketing.mobile.signal
+
+import com.adobe.marketing.mobile.Event
+import com.adobe.marketing.mobile.utils.DataReader
+
+// extends Event class represented as a rule consequence Event for Signal extension
+
+@JvmSynthetic
+internal fun Event.isPostback(): Boolean {
+    return this.consequenceType() == SignalConstants.EventDataKeys.Signal.RULES_RESPONSE_CONSEQUENCE_TYPE_POSTBACKS
+}
+
+@JvmSynthetic
+internal fun Event.isOpenUrl(): Boolean {
+    return this.consequenceType() == SignalConstants.EventDataKeys.Signal.RULES_RESPONSE_CONSEQUENCE_TYPE_OPEN_URL
+}
+
+@JvmSynthetic
+internal fun Event.isCollectPii(): Boolean {
+    return this.consequenceType() == SignalConstants.EventDataKeys.Signal.RULES_RESPONSE_CONSEQUENCE_TYPE_PII
+}
+
+@JvmSynthetic
+internal fun Event.contentType(): String {
+    return try {
+        DataReader.getString(this.details(), SignalConstants.EventDataKeys.RuleEngine.CONTENT_TYPE)
```

Comment:
nit: I would prefer using optString method everywhere

Reply:
https://github.com/adobe/aepsdk-core-android/blob/09dabcd1b19215f21c68698ffdd9cd779b49d480/code/android-core-library/src/main/java/com/adobe/marketing/mobile/utils/DataReader.java#L135
DataReader throws a runtime exception, so I have to do try-catch here. I can address this comment later after changing the above logic.